pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -142,16 +140,19 @@ func lookupDomainInterfaceByName(domainIfaces []api.Interface, networkName strin
 }
 
 func networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi *v1.VirtualMachineInstance, indexedDomainIfaces map[string]api.Interface) []v1.Network {
+	ifacesToHotplug := netvmispec.FilterIfacesForHotplug(vmi.Spec.Domain.Devices.Interfaces)
+	pluggableIfaces := netvmispec.IndexInterfaceSpecByName(ifacesToHotplug)
+
 	var networksToHotplug []v1.Network
 	interfacesToHoplug := netvmispec.IndexInterfacesFromStatus(
 		vmi.Status.Interfaces,
 		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
-			_, exists := indexedDomainIfaces[ifaceStatus.Name]
```

Comment:
I think the same fix as was done in virt-handler should be applied here. Checking the info source.

Reply:
In virt-launcher context, the most accurate indication of whether an interface is plugged is the domain object, the status may not reflect the actual state at that expect point of time.

I am not sure why its necessary to check the status at all, its possible to attach an interface whether the external interface exist or not (add by the CNI), isn't it?

Reply:
I didn't mean you should remove the domain check. I meant using the info-source instead of checking the binding.

> I am not sure why its necessary to check the status at all, its possible to attach an interface whether the external interface exist or not (add by the CNI), isn't it?
No,  the hotplug is serial. The interface can be attached to the guest only after it was attached to the pod and the tap device was created.

Reply:
As we concluded on an offline discussion, checking the binding is crucial for the scenario SR-IOV interfaces exist, because the checking if interface exist in domain will fail because SR-IOV interfaces are backed by host-devices.

Reply:
For tracking history, please mention in the commit message that the change to the virt-handler status calculation didn't cause the issue this commit fixes. It was there before as well.

Reply:
Sorry I don't follow, what you mean by " the change to the virt-handler status calculation"?

Reply:
The first commit in this PR - `virt-handler, VMI iface status: Keep detached secondary ifaces status`.

I will clarify my comment - commit `virt-launcher, iface hotplug: Filter out unsupported ifaces` could be a standalone PR. The issue it fixes ("SyncVMI routine fails, causing virt-handler to keep re-enqueue the request.") wasn't caused by the first commit. I existed before.
Also the commit message says- "Migrated VMs with SR-IOV interfaces are also affected, their SR-IOV interfaces wont plug back after migration.". I don't think it is the case, I don't understand how the syncVMI can cause it.

Reply:
> I will clarify my comment - commit virt-launcher, iface hotplug: Filter out unsupported ifaces could be a standalone PR. The issue it fixes ("SyncVMI routine fails, causing virt-handler to keep re-enqueue the request.") wasn't caused by the first commit. I existed before.

It cannot be standalone because having this change requires filtering host-devices (i.e.: SR-IOV interfaces) at virt-launcher level on hot-un/plug flow, otherwise, virt-handler gets into re-enqueues storm;
because SyncVMI routine tries  to hot-un/plug SR-IOV interfaces and fails to fetch in Libevirt XML spec.

> Also the commit message says- "Migrated VMs with SR-IOV interfaces are also affected, their SR-IOV interfaces wont plug back after migration.". I don't think it is the case, I don't understand how the syncVMI can cause it.

Migrated VMs also affected due to what is described above, the regular hot-un/plug flow will try to plug SR-IOV interfaces whether they are plugged or not.